sdk/python/feast/client.py

```diff
@@ -363,6 +363,8 @@ def apply(self, feature_sets: Union[List[FeatureSet], FeatureSet]):
         Args:
             feature_sets: List of feature sets that will be registered
         """
+        if self.project is not None and self.project not in self.list_projects():
```

Comment:
Although this technically work for the Python SDK, it won't cover the Feast Core side. Some of our users interface directly with the gRPC API. So the creation should ideally happen on that side.

Reply:
@woop agree that this should not be the responsibility of the client. Since I still need to update the tests to mock the `list_projects` response,  do you see this as an appropriate solution for now?

Perhaps going forward, it would be more appropriate to create the `default` project at the moment of building Feast Core?

Reply:
Sure, I think its ok for now. We could move it to Feast Core later.

Reply:
I think it would be safer to just create a project in Feast Core whenever a new project name comes in. Moving the `default` project string to Feast Core later makes sense as well, but I am happy to leave that in the Python SDK for now.